src/contracts/libraries/GPv2SafeERC20.sol

```diff
@@ -29,36 +39,55 @@ library GPv2SafeERC20 {
-            token,
-            abi.encodeWithSelector(token.transferFrom.selector, from, to, value)
-        );
-    }
+        bytes4 selector_ = token.transferFrom.selector;
 
-    /// @dev A reimplementation of the function with the same name in
-    /// Openzeppelin's SafeERC20. Unlike Openzeppelin's implementation, this
-    /// function does not revert if an address without code is called.
-    ///
-    /// @param token The token targeted by the call.
-    /// @param data The call data (encoded using abi.encode or one of its
-    /// variants).
-    function _callOptionalReturn(IERC20 token, bytes memory data) private {
-        // solhint-disable-next-line avoid-low-level-calls
-        (bool success, bytes memory returndata) = address(token).call(data);
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            let freeMemoryPointer := mload(0x40)
+            mstore(freeMemoryPointer, selector_)
+            mstore(add(freeMemoryPointer, 4), from)
+            mstore(add(freeMemoryPointer, 36), to)
+            mstore(add(freeMemoryPointer, 68), value)
 
-        if (!success) {
-            // Assembly used to revert with correctly encoded error message.
-            // solhint-disable-next-line no-inline-assembly
-            assembly {
-                revert(add(returndata, 0x20), mload(returndata))
+            if iszero(call(gas(), token, 0, freeMemoryPointer, 100, 0, 0)) {
+                returndatacopy(0, 0, returndatasize())
+                revert(0, returndatasize())
             }
         }
-        if (returndata.length > 0) {
-            // Return data is optional
-            require(
-                abi.decode(returndata, (bool)),
-                "GPv2SafeERC20: failed transfer"
-            );
+
+        require(getLastTansferResult(), "GPv2: failed transferFrom");
+    }
+
+    /// @dev Verifies that the last return was a successful `transfer*` call.
+    /// This is done by checking that the return data is either empty, or
+    /// is a valid ABI encoded boolean.
+    function getLastTansferResult() private pure returns (bool success) {
+        // NOTE: Inspecting previous return data requires assembly. Note that
+        // we write the return data to memory 0 in the case where the return
+        // data size is 32, this is OK since the first 64 bytes of memory are
+        // reserved by Solidy as a scratch space that can be used within
+        // assembly blocks.
+        // <https://docs.soliditylang.org/en/v0.7.6/internals/layout_in_memory.html>
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            switch returndatasize()
+                // Non-standard ERC20 transfer without return.
+                case 0 {
+                    success := 1
+                }
+                // Standard ERC20 transfer returning boolean success value.
+                case 32 {
+                    returndatacopy(0, 0, returndatasize())
```

Comment:
```suggestion
                    returndatacopy(0, 0, 32)
```

Reply:
Makes it nicer to read. The reason I had it this way is that `returndatasize()` costs 2 gas and `push 32` costs 3 and generates larger code.

Reply:
Nice trick! Then let's keep it as it is.